tests/System.Text.Encodings.Web.Utf8.Tests/UnescapeUrlPathInPlaceTests.cs

```diff
@@ -0,0 +1,18 @@
+﻿using Xunit;
+
+namespace System.Text.Encodings.Web.Utf8.Tests
+{
+    public class UnescapeUrlPathInPlaceTests : UrlEncoderTests
+    {
+        protected override void TestCore(string raw, string expected)
+        {
+            var input = GetBytes(raw);
+
+            var len = UrlEncoder.UnescapeInPlace(input);
+            Assert.True(len <= input.Length);
+
+            var outputDecoded = Encoding.UTF8.GetString(input.Slice(0, len).ToArray());
```

Comment:
Skip for now, the outcome of `Utf8String.ToString` is inconsistent with `Encoding.UTF8.GetString`. It may be an issue in `Utf8String`.